packages/nimble-components/src/table-column/base/cell-view/index.ts

```diff
@@ -18,10 +23,30 @@ export abstract class TableCellView<
     @observable
     public columnConfig!: TColumnConfig;
 
+    public column?: TableColumn<TColumnConfig>;
+
     /**
      * Called if an element inside this cell view has focus, and this row/cell is being recycled.
      * Expected implementation is to commit changes as needed, and blur the focusable element (or close
      * the menu/popup/etc).
      */
     public focusedRecycleCallback(): void {}
+
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        if (!this.column) {
+            return;
+        }
+        for (const delegatedEvent of this.column.columnInternals
+            .delegatedEvents) {
+            this.addEventListener(delegatedEvent, (event: Event) => {
+                this.column?.dispatchEvent(
+                    new CustomEvent<DelegatedEventEventDetails>(
+                        'delegated-event',
+                        { detail: { originalEvent: event } }
+                    )
+                );
+            });
+        }
+    }
```

Comment:
This is making too many assumptions about property assignment order. It is assuming the column property is already assigned before connectedCallback runs. And I have no idea how this is suppossed to behave if column isn't defined before connected. It just short circuits and returns early without an error and just never fires delegated events.

Always assume properties are dynamic and elements are dynamically attached and removed.
Change column to be observable and on columnChanged unregister existing handlers and register new ones.
Create tests for the cell view element to exercise this

Reply:
Created tests that changing the column property updates the event handlers properly.